components/scream/src/physics/share/physics_test_data.hpp

```diff
@@ -0,0 +1,125 @@
+#ifndef SCREAM_PHYSICS_TEST_DATA_HPP
+#define SCREAM_PHYSICS_TEST_DATA_HPP
+
+#include "share/scream_types.hpp"
+
+#include "ekat/util/ekat_math_utils.hpp"
+#include "ekat/ekat_assert.hpp"
+
+#include <vector>
+#include <utility>
+
+//
+// PhysicsTestData is meant to offer the client something they can inherit to provide
+// convenient handling of arrays of data in the common *Data structs that are used for
+// unit-testing and bridging. This class supports up to 4 classes of data
+//   * dim1 of Real
+//   * (dim1 x dim2) of Real
+//   * (dim1 x dim3) of Real
+//   * dim1 of Int
+//
+// Subclasses of PhysicsTestData should look like the following:
+/*
+struct SHOCGridData : public PhysicsTestData {
+  // Inputs
+  Real *zt_grid, *zi_grid, *pdel;
+
+  // In/out
+  Real *dz_zt, *dz_zi, *rho_zt;
+
+  SHOCGridData(Int shcol_, Int nlev_, Int nlevi_) :
+    PhysicsTestData(shcol_, nlev_, nlevi_,
+      {&zt_grid, &dz_zt, &pdel, &rho_zt},  // list of (shcol x nlev) members
+      {&zi_grid, &dz_zi}) {}               // list of (shcol x nlevi) members
+
+  SHOCGridData(const SHOCGridData &rhs) : PhysicsTestData(rhs,
+    {&zt_grid, &dz_zt, &pdel, &rho_zt, &zi_grid, &dz_zi}) {} // list of all PhysicsTestData managed Real members
+    // order of these members should match the order they appeared in the constructor
+
+  SHOCGridData &operator=(const SHOCGridData &rhs) { PhysicsTestData::operator=(rhs); return *this; }
+};
+*/
+
+namespace scream {
+
+// Base class for common test data setup
+struct PhysicsTestData
+{
+  Int shcol, nlev, nlevi;
```

Comment:
I think it's either "shared columns" or "shoc columns". It's the same concept as "ni" in P3 I think.

Reply:
If that's the case, then maybe we should change the name? Shared columns makes little sense to me (shared with whom, and in what sense?). And if it is "shoc_columns", then it's also not great, since it's in the common physics package (and what if we replace shoc with something else down the road?).

I would suggest something obvious, like `nlev`, which is the analogue of `nlev`. Uniform/consistent naming can help understand code better/faster, imho.

I understand that there are many files affected, but since this PR is already disruptive, a name change like that is probably best done here than in a later PR, so one only has conflicts once.

Reply:
I agree. I'll do this in a subsequent PR.